editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -5225,42 +5225,6 @@ void SpatialEditor::_unhandled_key_input(Ref<InputEvent> p_event) {
 		return;
 
 	snap_key_enabled = Input::get_singleton()->is_key_pressed(KEY_CONTROL);
-
-	Ref<InputEventKey> k = p_event;
-
-	if (k.is_valid()) {
-
-		// Note: need to check is_echo because first person movement keys might still be held
-		if (!is_any_freelook_active() && !p_event->is_echo()) {
```

Comment:
This doesn't cause issues with freelook?

Reply:
From quick testing, it seems to work fine, both when using freelook by holding the right mouse button or by pressing <kbd>Shift+F</kbd> then moving around.